addons/modules/functions/fnc_moduleNukeLocal.sqf

```diff
@@ -0,0 +1,382 @@
+#include "script_component.hpp"
+/*
+ * Author: Bohemia Interactive, Moerderhoschi, mharis001
+ * Zeus module function to detonate a nuclear bomb.
+ *
+ * Arguments:
+ * 0: Position ASL <ARRAY>
+ * 1: Destruction Radius <NUMBER>
+ * 2: Color Corrections <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_position, 1000, true] call zen_modules_fnc_moduleNukeLocal
+ *
+ * Public: No
+ */
+
+params ["_position", "_destructionRadius", "_colorCorrections"];
+
+// Destruction effects, only handled on the server
+if (isServer) then {
+    {
+        if (isDamageAllowed _x && {!(_x isKindOf "Logic")} && {!(_x isKindOf "VirtualMan_F")}) then {
+            private _delay = linearConversion [0, _destructionRadius, _x distance2D _position, 0, 20, true];
```

Comment:
A bigger delay might be useful, at 1km radius the impact of levelling a forest is a frozen game for at least a minute.

Reply:
I have now been waiting for at least three minutes for my game to unfreeze.

Reply:
Maybe increase the delay with more objects.